### PR TITLE
Allow all countries for phone

### PR DESCRIPTION
**Pull request by jeppesens**

### CODE REVIEW

File: validator/phone_number.py [hunk 1]
```diff
@@ -32,19 +32,23 @@ def _validate_and_normalize(p: str, cc: str = None, _try_again: bool = True) ->
...
-                raise _exception()
-            return p
+        logging.warning('Validating phone_number without knowing country!')
+
+        def safe_validator(p: str, cc: str) -> Union[str, Literal[False]]:
+            try:
+                return validate_and_normalize(p, cc)
+            except Exception:
+                return False
+        p = next(safe_validator(phone_number, cc) for cc in Config.countries)
```
Comment (pn) [inline validator/phone_number.py]: Isn't this checking only SE?

**Review by pn — COMMENTED** [review]
I assume this is for supporting numbers that have country code within number already? Stating reason we have a PR in the description helps reviewing. It would be also nice to have unit tests 🙂

File: validator/phone_number.py (hunk 1 above)
Comment (jeppesens) [reply validator/phone_number.py]: Nice catch, and should be fixed now

File: validator/phone_number.py [hunk 2]
```diff
@@ -46,9 +46,10 @@ def safe_validator(p: str, cc: str) -> Union[str, Literal[False]]:
...
             except Exception:
                 return False
-        p = next(safe_validator(phone_number, cc) for cc in Config.countries)
-        if not p:
-            raise _exception()
-        return p
+        for cc in Config.countries:
+            p = safe_validator(phone_number, cc)
+            if p:
+                return p
```
Comment (pn) [inline validator/phone_number.py:52]: I don't like this logic that much. If someone makes a typo in their phone number the phone number might be interpreted as a number in different country and be actually invalid (or vice versa). I think we should provide context in which country we validate phone number and only try to apply that prefix (maybe that's the purpose of `country_code`). So either someone provides country prefix or we use default one for current country (if we know one).

Comment (jeppesens) [reply validator/phone_number.py:52]: Exactly, so either you send country or the service guesses what country it is.

Comment (pn) [reply validator/phone_number.py:52]: @jeppesens, but if you have SE number and make a typo it could appear as valid number in NO 😬

Comment (jeppesens) [reply validator/phone_number.py:52]: But we don't have much choice here do we ?

Comment (pn) [reply validator/phone_number.py:52]: We have choice ro remove this logic. But maybe there is something I don't understand why we may want to keep it.

**Pull request merged**